### ISSUE TITLE
WNormLoss

### ISSUE THREAD

**Issue by hhuyujie** [eladrich/pixel2style2pixel#49]
Thanks for your open code. I want to konw how the WNormLoss work in the training, and if I don't use it, it will or not influence the results?

**Comment by yuval-alaluf** [comment]
The WNormLoss can be viewed as a form of regularization leading the encoder to output latent style vectors that are closer to the average latent vector. This is similar to the truncation trick introduced in the StyleGAN paper.  
Regarding the use of the loss - this depends on the task at hand. For encoding, we do not use the regularization loss. However, for our other applications we use a small weight for this loss. We found that this loss improves the quality of the images in these applications.

**Comment by Hzzone** [comment]
I disagree that WNormLoss can be seen as the truncation trick of StyleGAN. In my view, the right form of WNormLoss should be written as:
L_reg = E(x) **2;
Specifically, the outputs of the encoder should not be very large, otherwise, E(x)+\bar{w} will be located out of the W space of StyleGAN generator. In a sense, I think WNormLoss should play a role like weight decay that restricts the space of the encoder.
